### PR TITLE
Provider: Gcore

### LINKED ISSUE #12: desec & gcore providers
The domain i'm using is hosted on desec and gcore.
Is it possible to implement these providers?
Thank you for sharing this operator!

### PR DESCRIPTION
**Pull request by pier-oliviert**
Here's the PR to add support for [GCore](gcore.com). I added a few things around like cleanup that should have been in a different PR, but I realized it just now.

This should close #12

### CODE REVIEW

File: pkg/providers/gcore/client.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	"fmt"
+	"net/url"
+	"os"
+
+	gdns "github.com/G-Core/gcore-dns-sdk-go"
+	phonebook "github.com/pier-oliviert/phonebook/api/v1alpha1"
+)
+
+const (
+	EnvAPIURL   = "GCORE_API_URL"
```
Comment (ainsey11) [inline pkg/providers/gcore/client.go:14]: Does this need adding to the docs and samples or does it have a default set that I can't spot?

File: pkg/providers/gcore/client.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	zones       []string
+	api         api
+}
+
+func NewClient(ctx context.Context) (*gcore, error) {
+	var err error
+
+	apiURL := os.Getenv(EnvAPIURL)
+
+	token := os.Getenv(EnvAPIToken)
```
Comment (ainsey11) [inline pkg/providers/gcore/client.go:38]: Should these be set from the normal method? Apologies if I've missed it (on my phone ATM and GitHub isn't the greatest 😂) but I've been using the `utils.RetrieveValueFromEnvOrFile` function to pull these in

**Review by ainsey11 — COMMENTED** [review]
Looks pretty good!

I've added a couple of minor comments on, tests and everything looks good to me too, 

Nice work! 👌

File: pkg/providers/gcore/client.go (hunk 2 above)
Comment (pier-oliviert) [reply pkg/providers/gcore/client.go:38]: Honestly, I copied these from https://github.com/G-Core/external-dns-gcore-webhook/blob/master/gcoreprovider/gcore.go with no second thoughts. I'll make the changes to use `utils.RetrieveValueFromEnvOrFile`.

File: pkg/providers/gcore/client.go (hunk 1 above)
Comment (ainsey11) [reply pkg/providers/gcore/client.go:14]: Told you I was blind! 

Sounds like a plan to me 😃

**Review by ainsey11 — APPROVED** [review]
Just marking as approved before I head to bed knowing you've already fed back on my comments :)

**Pull request merged**